storage/retrieval.go

```diff
@@ -106,7 +110,25 @@ func getFirstTestChunkAfter(allTestChunks []LogChunkInfo, target time.Time) time
 	return TimeRangeMax
 }
 
-func (storage *Bucket) GetTestLogLines(context context.Context, buildId string, testId string) (LogIterator, error) {
+func channelFromIterator(context context.Context, iterator LogIterator) chan *model.LogLineItem {
+	logsChan := make(chan *model.LogLineItem)
+
+	go func() {
+		defer close(logsChan)
+		for iterator.Next(context) {
+			if iterator.Err() != nil {
+				grip.Errorf("Error iterating over logs: %v", iterator.Err())
+				break
```

Comment:
It looks like it will return False for next on an unrecoverable error, so we can just defer and grab/log the errors at the end which just calls catcher.Resolve().